identity/src/main/java/com/android/identity/util/SimpleApplicationData.java

```diff
@@ -0,0 +1,252 @@
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return new String(value, StandardCharsets.UTF_8);
+    }
+
+    @Nullable
+    @Override
+    public Long getApplicationDataNumber(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return Util.cborDecodeLong(value);
+    }
+
+    @Nullable
+    @Override
+    public Timestamp getApplicationDataTimestamp(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return Timestamp.ofEpochMilli(Util.cborDecodeLong(value));
+    }
+
+    @Nullable
+    @Override
+    public Boolean getApplicationDataBoolean(@NonNull String key) {
+        byte[] value = mApplicationData.get(key);
+        if (value == null) {
+            return null;
+        }
+        return Util.cborDecodeBoolean(value);
+    }
+
+    @Override
+    public byte[] serialize() {
+        CborBuilder appDataBuilder = new CborBuilder();
+        MapBuilder<CborBuilder> appDataMapBuilder = appDataBuilder.addMap();
+        for (String key : mApplicationData.keySet()) {
+            appDataMapBuilder.put(key, mApplicationData.get(key));
+        }
+        appDataMapBuilder.end();
+        return Util.cborEncode(appDataBuilder.build().get(0));
+    }
+
+    /**
+     * Returns a fully populated SimpleApplicationData from a CBOR-encoded byte[].
+     *
+     * <p> To encode a SimpleApplicationData, use {@link #serialize()}.
+     *
+     * @param encodedApplicationData The byte array resulting from {@link #serialize()}.
+     * @return A SimpleApplicationData with the correct key, value pairs.
+     */
+    public static SimpleApplicationData deserialize(byte[] encodedApplicationData) {
+        ByteArrayInputStream bais = new ByteArrayInputStream(encodedApplicationData);
+        List<DataItem> dataItems;
+        try {
+            dataItems = new CborDecoder(bais).decode();
+        } catch (CborException e) {
+            throw new IllegalStateException("Error decoded CBOR", e);
+        }
+        if (dataItems.size() != 1) {
+            throw new IllegalStateException("Expected 1 item, found " + dataItems.size());
+        }
+        if (!(dataItems.get(0) instanceof co.nstant.in.cbor.model.Map)) {
+            throw new IllegalStateException("Item is not a map");
+        }
+        DataItem applicationDataDataItem = dataItems.get(0);
+
+        SimpleApplicationData appData = new SimpleApplicationData();
+        for (DataItem keyItem : ((co.nstant.in.cbor.model.Map) applicationDataDataItem).getKeys()) {
+            String key = ((UnicodeString) keyItem).getString();
+            byte[] value = Util.cborMapExtractByteString(applicationDataDataItem, key);
+            appData.setApplicationData(key, value);
```

Comment:
This works but is a bit overhead (checking if callbacks should be fired etc.). Just do `mApplicationData.put(key, value);` instead

Reply:
This method is a static method which makes new `SimpleApplicationData` objects, so I can't just add to the `mApplicationData` directly